### PR TITLE
Support for Next 14

### PR DESCRIPTION
**Pull request by arakelovd**
Adding support for Next@14.

### CODE REVIEW

**Comment by martinnabhan** [comment]
Awesome, thank you!
I've added some fixes and turns out Next.js 14 drops support for Node.js 16 so we'll also have to drop Node.js 16.

@trevor-scheer Could you please take a look before I merge?

File: tsconfig.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "esModuleInterop": true,
     "module": "commonjs",
     "outDir": "./dist",
+    "skipLibCheck": true,
```
Comment (martinnabhan) [inline tsconfig.json:10]: Upgrading `next` to 14 causes a type mismatch for `next12`, which is why I've disabled lib check.

Comment (arakelovd) [reply tsconfig.json:10]: Thank you, this all looks great.

**Comment by karl-run** [comment]
I can verify that the integration itself works with Next 14 on my projects.

**Review by marcobaro1991 — APPROVED** [review]

**Comment by arakelovd** [comment]
@trevor-scheer  can we get some eyes on this please?

File: src/__tests__/integration.test.ts [hunk 2]
```diff
@@ -7,20 +7,35 @@ import {
...
+      ({ apiResolver } = await import('next12/dist/server/api-utils/node'));
+      break;
+    case '13':
+      ({ apiResolver } = await import('next13/dist/server/api-utils/node/api-resolver'));
+      break;
+    case '14':
+      ({ apiResolver } = await import('next14/dist/server/api-utils/node/api-resolver'));
+      break;
+    default:
+      throw new Error('Next.js version not specified.');
```
Comment (trevor-scheer) [inline src/__tests__/integration.test.ts:24]:
```suggestion
      throw new Error('Next.js version not specified. Ensure the Next.js version is specified via the NEXT_VERSION environment variable.');
```

File: tsconfig.json (hunk 1 above)
Comment (trevor-scheer) [reply tsconfig.json:10]: Alternatively, dropping support for Next 12 would be reasonable since this is a major release but I agree with this decision.

**Review by trevor-scheer — DISMISSED** [review]

**Review by martinnabhan — APPROVED** [review]

**Pull request merged**